### PR TITLE
provide better error handling logic for function authorization when a tenant doesn't exist

### PR DESCRIPTION
**Pull request by jerrypeng**
### Motivation

Return 403 instead of 500

### CODE REVIEW

**Comment by jerrypeng** [comment]
rerun java8 tests

**Comment by jerrypeng** [comment]
rerun integration tests

File: pulsar-functions/worker/src/main/java/org/apache/pulsar/functions/worker/rest/api/ComponentImpl.java [hunk 1]
```diff
@@ -1694,9 +1694,13 @@ public boolean isAuthorizedRole(String tenant, String namespace, String clientRo
...
-                TenantInfo tenantInfo = worker().getBrokerAdmin().tenants().getTenantInfo(tenant);
-                if (tenantInfo.getAdminRoles() != null && tenantInfo.getAdminRoles().contains(clientRole)) {
-                    return true;
+                try {
+                    TenantInfo tenantInfo = worker().getBrokerAdmin().tenants().getTenantInfo(tenant);
+                    if (tenantInfo != null && tenantInfo.getAdminRoles() != null && tenantInfo.getAdminRoles().contains(clientRole)) {
+                        return true;
+                    }
+                } catch (PulsarAdminException.NotFoundException e) {
+
```
Comment (srkukarni) [inline pulsar-functions/worker/src/main/java/org/apache/pulsar/functions/worker/rest/api/ComponentImpl.java]: dont really understand whats the intention here? if the tenant doesn't exist, won';t it fail in the next call for getting namespaces in that tenant?

Comment (jerrypeng) [reply pulsar-functions/worker/src/main/java/org/apache/pulsar/functions/worker/rest/api/ComponentImpl.java]: are you referring ```NamespaceName.get(tenant, namespace)```?  That just returns the correctly formatted name of the namespace

Comment (srkukarni) [reply pulsar-functions/worker/src/main/java/org/apache/pulsar/functions/worker/rest/api/ComponentImpl.java]: Got it

**Comment by jerrypeng** [comment]
rerun java8 tests

**Pull request merged**